feat(openpipeline): Adds acceptance test examples

d0weinberger opened the pull request:
#### **Why** this PR and **What** has changed?
This PR adds acceptance tests for all OpenPipelineV2 schemas.

#### **How** does it do it?
I started out with configs for the `events` data type.
It contains two example configs for ingest sources:
- One config with a minimal number of attributes set
- One with everything set that you can possibly set, and all different kinds of processors for the pre-processing step.

It also contains two example configs for pipelines:
- One config with a minimal number of attributes set
- One with everything set that you can possibly set, and all different kinds of processors for the pre-processing step.

It also contains one example config for routing, which contains two entries:
 - One mapping to the builtin default pipeline
 - One mapping to a custom pipeline, defined in the same example. (I had to remove this one because of a bug in the server when handling `repairInput`. Will add it again, as soon as it's resolved).
 
I wanted all of the configs to make sense and I tried them all out. Below is a screenshot of the ingest source example with everything set.

<img width="1317" height="761" alt="image" src="https://github.com/user-attachments/assets/6c644f10-b527-4343-b1f1-8171cd93468d" />

Test files are put in `testdata/terraform` and a `service_test.go` is created, following the example of all the other acceptance test files.

Afterwards I copy-pasted these files for all the other schemas there are, with a few necessary adaptions:
- Some features were not yet available, so I skipped the tests
- I skipped ingestsource tests for data types where custom ingest sources are not (typically) allowed.
- I adapted the processors, as not all stages are allowed for all data types.

#### How does it affect **users**?
It doesn't

**Issue:**
CA-15949

d0weinberger committed:
feat(openpipeline): Adds ingest source examples for events.ingestsources

d0weinberger committed:
feat(openpipeline): Adds pipeline examples for events.pipelines

d0weinberger committed:
feat(openpipeline): Adds routing examples for events.routing

d0weinberger committed:
feat(openpipeline): Removes stages cost-allocation and product-allocation

These stages are not supported yet on PROD and they fail our acceptance tests. Re-adding them as soon as the stages are supported on PROD

d0weinberger committed:
feat(openpipeline): Removes routing table example with settings ID for now

Currently this does not work due to the way repairInput is handled on server-side. Will re-add it later

d0weinberger committed:
feat(openpipeline): Copy-paste acceptance tests to all OpenPipeline data types

d0weinberger committed:
feat(openpipeline): Skips tests for not-yet-enabled features

d0weinberger committed:
feat(openpipeline): Skips ingest source tests for schemas which don't allow custom ingest sources

d0weinberger committed:
feat(openpipeline): Adapts pipeline for schemas which don't allow certain processors

arthurpitman commented on dynatrace/api/builtin/openpipeline/azure/logs/forwarding/routing/testdata/terraform/example.tf:
```diff
@@ -0,0 +1,11 @@
+      description         = "Default route"
+    }
+  }
+}
```
There's a missing newline here and at the end of many other `.tf` files

d0weinberger replied:
Fixed

arthurpitman commented on dynatrace/api/builtin/openpipeline/davis/events/routing/testdata/terraform/example.tf:
```diff
@@ -0,0 +1,11 @@
+  routing_entries {
+    routing_entry {
+      enabled             = true
+      pipeline_type       = "builtin"
```
Is it actually possible to configure anything here? In the previous OpenPipeline configuration, we removed all the built-in entries as no configuration was possible.

d0weinberger replied:
Yes, there could be more than one builtin pipeline (e.g. for security events that is the case), so the ability to select the specific pipeline via the ID is important.

d0weinberger committed:
feat(openpipeline): Fixes newlines in example files

Kirdock commented on dynatrace/api/builtin/openpipeline/events/sdlc/ingestsources/testdata/terraform/maximal-example.tf:
```diff
@@ -0,0 +1,75 @@
+resource "dynatrace_openpipeline_v2_events_ingestsources" "maximal-source" {
```
```suggestion
resource "dynatrace_openpipeline_v2_events_sdlc_ingestsources" "maximal-source" {
```

Kirdock commented on dynatrace/api/builtin/openpipeline/events/sdlc/ingestsources/testdata/terraform/minimal-example.tf:
```diff
@@ -0,0 +1,6 @@
+resource "dynatrace_openpipeline_v2_events_ingestsources" "minimal-source" {
```
```suggestion
resource "dynatrace_openpipeline_v2_events_sdlc_ingestsources" "minimal-source" {
```

Kirdock commented on dynatrace/api/builtin/openpipeline/events/sdlc/pipelines/testdata/terraform/maximal-example.tf:
```diff
@@ -0,0 +1,205 @@
+resource "dynatrace_openpipeline_v2_events_pipelines" "max-pipeline" {
```
```suggestion
resource "dynatrace_openpipeline_v2_events_sdlc_pipelines" "max-pipeline" {
```

Kirdock commented on dynatrace/api/builtin/openpipeline/events/sdlc/pipelines/testdata/terraform/minimal-example.tf:
```diff
@@ -0,0 +1,12 @@
+resource "dynatrace_openpipeline_v2_events_pipelines" "min-pipeline" {
```
```suggestion
resource "dynatrace_openpipeline_v2_events_sdlc_pipelines" "min-pipeline" {
```

Kirdock commented on dynatrace/api/builtin/openpipeline/events/sdlc/routing/testdata/terraform/example.tf:
```diff
@@ -0,0 +1,11 @@
+resource "dynatrace_openpipeline_v2_events_routing" "routing" {
```
```suggestion
resource "dynatrace_openpipeline_v2_events_sdlc_routing" "routing" {
```

Kirdock commented on dynatrace/api/builtin/openpipeline/events/security/ingestsources/testdata/terraform/maximal-example.tf:
```diff
@@ -0,0 +1,75 @@
+resource "dynatrace_openpipeline_v2_events_ingestsources" "maximal-source" {
```
```suggestion
resource "dynatrace_openpipeline_v2_events_security_ingestsources" "maximal-source" {
```

Kirdock commented on dynatrace/api/builtin/openpipeline/events/security/ingestsources/testdata/terraform/minimal-example.tf:
```diff
@@ -0,0 +1,6 @@
+resource "dynatrace_openpipeline_v2_events_ingestsources" "minimal-source" {
```
```suggestion
resource "dynatrace_openpipeline_v2_events_security_ingestsources" "minimal-source" {
```

Kirdock commented on dynatrace/api/builtin/openpipeline/events/security/pipelines/testdata/terraform/maximal-example.tf:
```diff
@@ -0,0 +1,205 @@
+resource "dynatrace_openpipeline_v2_events_pipelines" "max-pipeline" {
```
```suggestion
resource "dynatrace_openpipeline_v2_events_security_pipelines" "max-pipeline" {
```

Kirdock commented on dynatrace/api/builtin/openpipeline/events/security/pipelines/testdata/terraform/minimal-example.tf:
```diff
@@ -0,0 +1,12 @@
+resource "dynatrace_openpipeline_v2_events_pipelines" "min-pipeline" {
```
```suggestion
resource "dynatrace_openpipeline_v2_events_security_pipelines" "min-pipeline" {
```

Kirdock commented on dynatrace/api/builtin/openpipeline/events/security/routing/testdata/terraform/example.tf:
```diff
@@ -0,0 +1,11 @@
+resource "dynatrace_openpipeline_v2_events_routing" "routing" {
```
```suggestion
resource "dynatrace_openpipeline_v2_events_security_routing" "routing" {
```

Kirdock commented on dynatrace/api/builtin/openpipeline/system/events/pipelines/testdata/terraform/maximal-example.tf:
```diff
@@ -0,0 +1,208 @@
+  custom_id = "pipeline_Warning_pipeline_2773_tf_#name#"
+  processing {
+    # processing is not available for system_events pipelines
+    # processors {
```
Could you remove the commented-out code? The comment above is fine

Kirdock commented on dynatrace/api/builtin/openpipeline/user/events/pipelines/testdata/terraform/maximal-example.tf:
```diff
@@ -0,0 +1,208 @@
+  custom_id = "pipeline_Warning_pipeline_2773_tf_#name#"
+  processing {
+    # processing is not available for system_events pipelines
+    # processors {
```
Could you remove the commented-out code? The comment above is fine.
Also for the davis field below

Kirdock commented on .github/workflows/tests.yml:
```diff
@@ -4511,6 +4511,510 @@ jobs:
+          DT_CLIENT_ID: ${{ secrets.DT_CLIENT_ID }}
+          DT_ACCOUNT_ID: ${{ secrets.DT_ACCOUNT_ID }}
+          DT_CLIENT_SECRET: ${{ secrets.DT_CLIENT_SECRET }}
+        run: go test -v ./dynatrace/api/builtin/openpipeline/azure/logs/forwarding/ingestsources
```
idea: should we combine all pipeline resources into `go test -v ./dynatrace/api/builtin/openpipeline`

arthurpitman replied:
I think it is a good idea

Kirdock reviewed (commented):
off topic: Do we have some outdated schemas? I'm just seeing `events_security` and also `security_events` resources for openpipeline. So I'm not sure if this is intended or not.

Edit: It's intended.

Kirdock committed:
feat(openpipeline): removed commented out HCL

Kirdock committed:
feat(openpipeline): fix resource names for examples

Some examples had "dynatrace_openpipeline_v2_events_..." instead of the actual one

Kirdock committed:
feat(openpipeline): merge all openpipeline test steps into one
